### PR TITLE
Feature/nested routing

### PR DESCRIPTION
**Pull request by szymonbrainhub**
Nested Routing:
Routes can be nested to inherit properties of its parent. 

Locals from response:
To route callback, as a third parameter, we are adding a locals extracted from response object.

README.md file updated. More details there

### CODE REVIEW

File: packages/hadron-demo/index.ts [hunk 1]
```diff
@@ -43,15 +43,17 @@ jsonProvider([`${__dirname}/routing/*`], ['config.js']).then((routes: any) => {
       hadronExpress,
     ],
     config,
-  ).then((container: IContainer) => {
-    expressApp.use((req, res, next) =>
-      res.status(404).json('Request not found.'),
-    );
-    container.register('customValue', 'From Brainhub with ❤️');
+  )
+    .then((container: IContainer) => {
```
Comment (piotr-s-brainhub) [inline packages/hadron-demo/index.ts:47]: maybe you could use `async / await` for a better readability

File: packages/hadron-express/README.md [hunk 2]
```diff
@@ -84,6 +84,22 @@ GET request with path: `http://localhost/Hello/World` will result with following
 "Hello World"
 ```
 
+### Locals (available from 2.0.0)
+
+As a third parameter, hadron delivers `locals` from your response. You can inject it's content in middlewares, f.e.
```
Comment (piotr-s-brainhub) [inline packages/hadron-express/README.md]: a typo - `it's` should be `its`

`f.e.` - I don't think such an abbreviation exists, it should be `e.g.`

File: packages/hadron-express/src/__tests__/hadronToExpress.ts [hunk 3]
```diff
@@ -131,15 +126,19 @@ describe('router config', () => {
       return routesToExpress(testRoute, Container).then(() =>
         request(app)
           .get(`/index`)
-          .then(() => {
-            expect(eventManager.emitEvent.calledOnce).to.equal(true);
-          }),
+          .then(() =>
```
Comment (piotr-s-brainhub) [inline packages/hadron-express/src/__tests__/hadronToExpress.ts:129]: especially here I think `async / await ` should improve much the readability

File: packages/hadron-express/src/hadronToExpress.ts [hunk 4]
```diff
@@ -87,22 +91,118 @@ const convertToExpress = (
...
+        .map((childKey: string) =>
+          prepareRoute(
+            (route.routes as any)[childKey],
+            childKey,
+            container,
+            preparedRoute,
+            routeKey,
+          ),
+        )
+        .reduce((agg: any, current: any) => ({ ...agg, ...current })),
```
Comment (piotr-s-brainhub) [inline packages/hadron-express/src/hadronToExpress.ts]: I'm for using whole words as variable names with only exceptions being very widely used ones like `req`, `res`

IMO `agg` should be named `accumulator` (or `aggregator` if you want)

File: packages/hadron-express/src/hadronToExpress.ts [hunk 5]
```diff
@@ -87,22 +91,118 @@ const convertToExpress = (
...
+    };
+  }
+
+  return result;
+};
+
+export const prepareMiddlewares = (
+  middleware: any[],
+  $middleware: any[],
```
Comment (piotr-s-brainhub) [inline packages/hadron-express/src/hadronToExpress.ts:180]: what's the difference between `middleware` and `$middleware`?

File: packages/hadron-demo/index.ts (hunk 1 above)
Comment (szymonbrainhub) [reply packages/hadron-demo/index.ts:47]: I'd rather stick to then/catch. We are using them all over the project. Also, readability is very subjective topic

File: packages/hadron-express/src/__tests__/hadronToExpress.ts (hunk 3 above)
Comment (szymonbrainhub) [reply packages/hadron-express/src/__tests__/hadronToExpress.ts:129]: As above, I would rather not change all then/catch, but stick to one. But if You feel like it's a game changer, feel free to do such refactor.

File: packages/hadron-express/src/hadronToExpress.ts (hunk 5 above)
Comment (szymonbrainhub) [reply packages/hadron-express/src/hadronToExpress.ts:180]: README.md. Basically, those are middlewares that should overwrite parents middleware

File: packages/hadron-express/src/hadronToExpress.ts (hunk 4 above)
Comment (szymonbrainhub) [reply packages/hadron-express/src/hadronToExpress.ts]: It's more like esthetic change, but you are right.

**Comment by szymonbrainhub** [comment]
Okay, 2+ months is a heck of a long time to wait for merge.

**Pull request merged**